### PR TITLE
Deprecate public schema API

### PR DESCRIPTION
**Pull request by kgaillot**

### CODE REVIEW

**Comment by kgaillot** [comment]
@clumens, here are the last big schema changes

File: lib/common/schemas.c [hunk 1]
```diff
@@ -852,6 +843,22 @@ pcmk__validate_xml(xmlNode *xml_blob, const char *validation,
     return false;
 }
 
+/*!
+ * \internal
+ * \brief Validate XML using its configured schema (and send errors to logs)
+ *
+ * \param[in] xml  XML to validate
+ *
+ * \return TRUE if XML validates, otherwise FALSE
```
Comment (clumens) [inline lib/common/schemas.c:793]: I don't know how consistent we're being in our docstrings, but it actually returns `true` or `false`.

File: include/crm/common/xml_compat.h [hunk 2]
```diff
@@ -146,6 +146,28 @@ gboolean add_message_xml(xmlNode *msg, const char *field, xmlNode *xml);
...
+//! \deprecated Do not use
+gboolean validate_xml(xmlNode *xml_blob, const char *validation,
+                      gboolean to_logs);
+
+//! \deprecated Do not use
+gboolean validate_xml_verbose(const xmlNode *xml_blob);
+
+//! \deprecated Do not use
+gboolean cli_config_update(xmlNode **xml, int *best_version, gboolean to_logs);
 //! \deprecated Do not use
```
Comment (clumens) [inline include/crm/common/xml_compat.h:171]: This could use a blank line.

File: lib/common/schemas.c (hunk 1 above)
Comment (kgaillot) [reply lib/common/schemas.c:793]: whoops, thanks

File: include/crm/common/xml_compat.h (hunk 2 above)
Comment (kgaillot) [reply include/crm/common/xml_compat.h:171]: whoops, thanks

**Comment by kgaillot** [comment]
updated commits per review

**Review by clumens — APPROVED** [review]

**Pull request merged**